NEWS.md

```diff
@@ -32,6 +32,77 @@
     (either lexical or physical). This also means that calling `pl$Categorical`
     doesn't create a `DataType` anymore. All calls to `pl$Categorical` must be
     replaced by `pl$Categorical()` (#860).
+-   The conversion strategy between the POSIXct type without time zone attribute
+    and Polars datetime has been changed (#878).
+    `POSIXct` class vectors without a time zone attribute have UTC time internally
+    and is displayed based on the system's time zone. Previous versions of `polars`
+    only considered the internal value and interpreted it as UTC time, so the
+    time displayed as `POSIXct` and in Polars was different.
+
+    ```r
+    # polars 0.14.1
+    Sys.setenv(TZ = "Europe/Paris")
+    datetime = as.POSIXct("1900-01-01")
+    datetime
+    #> [1] "1900-01-01 PMT"
+
+    s = polars::as_polars_series(datetime)
+    s
+    #> polars Series: shape: (1,)
+    #> Series: '' [datetime[ms]]
+    #> [
+    #>  1899-12-31 23:50:39
+    #> ]
+
+    as.vector(s)
+    #> [1] "1900-01-01 PMT"
+    ```
+
+    Now the internal value is updated to match the displayed value.
+
+    ```r
+    # polars 0.15.0
+    Sys.setenv(TZ = "Europe/Paris")
+    datetime = as.POSIXct("1900-01-01")
+    datetime
+    #> [1] "1900-01-01 PMT"
+
+    s = polars::as_polars_series(datetime)
+    s
+    #> polars Series: shape: (1,)
+    #> Series: '' [datetime[ms]]
+    #> [
+    #>  1900-01-01 00:00:00
+    #> ]
+
+    as.vector(s)
+    #> [1] "1900-01-01 PMT"
+    ```
+
+    This update may cause errors when converting from Polars to `POSIXct` for non-existent
+    or ambiguous times. It is recommended to explicitly add a time zone before converting
+    from Polars to R.
+
+    ```r
+    Sys.setenv(TZ = "America/New_York")
+    ambiguous_time = as.POSIXct("2020-11-01 01:00:00")
+    ambiguous_time
+    #> [1] "2020-11-01 01:00:00 EDT"
+
+    pls = polars::as_polars_series(ambiguous_time)
+    pls
+    #> polars Series: shape: (1,)
+    #> Series: '' [datetime[ms]]
+    #> [
+    #>  2020-11-01 01:00:00
+    #> ]
+
+    ## This will be error!
+    # pls |> as.vector()
+
+    pls$dt$replace_time_zone("UTC") |> as.vector()
+    #> [1] "2020-11-01 01:00:00 UTC"
+    ```
```

Comment:
That's a nice explanation, thanks. Can you add it somewhere in the docs as well?

Reply:
Definitely this needs to be documented, but I don't think there is currently any documentation on type mapping (#425).
Any suggestions?

Reply:
This is quite general about the interaction between polars and R, so I think it could be stored in a section "Timezones and date-time" in `DataFrame_class`. In any case, you could add the location of the docs in the error message, e.g  ```Please use `ambiguous` [...]. See section blahblah in `?DataFrame_class` for more details```

Reply:
> This is quite general about the interaction between polars and R, so I think it could be stored in a section "Timezones and date-time" in `DataFrame_class`. In any case, you could add the location of the docs in the error message, e.g `` Please use `ambiguous` [...]. See section blahblah in `?DataFrame_class` for more details ``

I can't figure out how to capture a specific error on the Rust side and add a string. Do you know of any other places that do that?

Reply:
You could do some `grep()` on the error message in `unwrap()` on the R side to add this additional message if the error is about timezone

Reply:
Are there other places that do that? I think that approach is fragile and not ideal.

Reply:
No there are no other places that do that, and I agree it's not great. An alternative is to let the error message unchanged, and instead add a link to this section of `DataFrame_class` in each function that pull data in R (`to_list()`, etc.)

Reply:
Yes I think it's good, I just had a few comments